suite-native/module-device-settings/src/hooks/useChangeDeviceName/index.ts

```diff
@@ -0,0 +1,91 @@
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { useForm } from '@suite-native/forms';
+import { useTranslate } from '@suite-native/intl';
+import {
+    DeviceNameStackParamList,
+    DeviceNameStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import TrezorConnect from '@trezor/connect';
+import { EventType, analytics } from '@trezor/suite-analytics';
+
+import { deviceNameFormValidationSchema } from './DeviceNameFormSchema';
+
+export const MAX_LENGTH = 16;
+
+type NavigationProps = StackNavigationProps<
+    DeviceNameStackParamList,
+    DeviceNameStackRoutes.DeviceName
+>;
+
+export const useChangeDeviceName = () => {
+    const { translate } = useTranslate();
+    const navigation = useNavigation<NavigationProps>();
+    const device = useSelector(selectSelectedDevice);
+
+    const form = useForm({
+        validation: deviceNameFormValidationSchema(translate),
+        defaultValues: {
+            deviceName: device?.label || '',
+        },
+        mode: 'onChange',
+        reValidateMode: 'onChange',
+    });
+
+    const deviceNameError = form.formState.errors.deviceName;
+    const deviceNameValue = form.watch('deviceName');
+    const isMaxLengthReached = deviceNameValue.length >= MAX_LENGTH;
+    const isSubmittable =
+        form.formState.isValid && deviceNameValue.length <= MAX_LENGTH && !deviceNameError;
```

Comment:
```suggestion
    const isSubmittable =
        form.formState.isValid && deviceNameValue.length <= MAX_LENGTH;
```

Is `!deviceNameError` redundant since we checked for valid form state?

Reply:
I would find it more readable to have this hook where the form is used. Since it's not logic that we might want to reuse elsewhere too much, it's just moving it from the component where it's used. IMO it's better to have it where you use it, mostly again from DX, but it's not that important. If you disagree and think the form component would be too "bloated", feel free to ignore my comment.

Reply:
https://github.com/trezor/trezor-suite/pull/19684/commits/f912231189f29ade5224d7da3831a45085953ca2 

Good point, but as the hook encompasses all kinds of unique variables, such as MAX_LENGTH and form setup, I would leave it as is. It was a bit tricky to decide where everything should go based on the file’s responsibility, but I decided to keep it clear... all in one hook instead of splitting it.